src/test/kotlin/uk/gov/dluhc/printapi/messaging/SendApplicationToPrintMessageListenerIntegrationTest.kt

```diff
@@ -147,33 +79,48 @@ internal class SendApplicationToPrintMessageListenerIntegrationTest : Integratio
-            assertThat(saved).usingRecursiveComparison()
-                .withEqualsForType(instantEqualToRoundedSeconds, Instant::class.java)
-                .ignoringFields(
-                    "vacNumber",
-                    "printRequests.requestId",
-                    "printRequests.statusHistory.eventDateTime"
-                )
-                .ignoringFieldsMatchingRegexes(
-                    ".*id",
-                    ".*version",
-                    ".*createdBy",
-                    ".*dateCreated",
-                )
-                .isEqualTo(expected)
-            assertThat(saved.status).isEqualTo(Status.PENDING_ASSIGNMENT_TO_BATCH)
-            assertThat(saved.printRequests[0].requestId).containsPattern(Regex("^[a-f\\d]{24}$").pattern)
-            assertThat(saved.vacNumber).containsPattern(Regex("^[A-Za-z\\d]{20}$").pattern)
+            assertSavedCertificate(saved, expected)
         }
     }
 
-    fun withinSecond(actual: Long, epochSeconds: Long): Boolean {
-        val variance = 1
-        val lowerBound: Long = epochSeconds - variance
-        val upperBound: Long = epochSeconds + variance
-        return actual in lowerBound..upperBound
+    @Test
+    fun `should process message received on queue given certificate exists`() {
+        // Given
+        val ero = buildElectoralRegistrationOfficeResponse(
+            localAuthorities = mutableListOf(
+                buildLocalAuthorityResponse(),
+                buildLocalAuthorityResponse(contactDetailsWelsh = buildContactDetails())
+            )
+        )
+        val localAuthority = ero.localAuthorities[1]
+        val gssCode = localAuthority.gssCode
+        val certificate = buildCertificate(gssCode = gssCode, sourceType = SourceType.VOTER_CARD)
+        val expected = certificateRepository.save(certificate)
+
+        // Resend must reference the same application for original saved certificate to be updated
+        val payload = buildSendApplicationToPrintMessage(
+            gssCode = gssCode,
+            sourceType = SqsSourceType.VOTER_MINUS_CARD,
+            sourceReference = certificate.sourceReference!!,
+            supportingInformationFormat = LARGE_MINUS_PRINT
+        )
+        wireMockService.stubEroManagementGetEroByGssCode(ero, gssCode)
+
+        // add resend print request from processing new send to print request to expected certificate
+        with(payload) { expected.addPrintRequest(toPrintRequest(localAuthority, SupportingInformationFormat.LARGE_PRINT)) }
+
+        // When
+        sqsMessagingTemplate.convertAndSend(sendApplicationToPrintQueueName, payload)
+
+        // Then
+        await.atMost(5, SECONDS).untilAsserted {
+            wireMockService.verifyEroManagementGetEro(gssCode)
+            val response = certificateRepository.findAll()
+            assertThat(response).hasSize(1)
+            val saved = response[0]
+            assertThat(saved.printRequests).hasSize(2)
```

Comment:
Not strictly necessary, but does confirm the second print request is loaded on the certificate before the full expected certificate is compared.